### PR TITLE
Adds clocks

### PR DESCRIPTION
**Pull request by MaxIsJoe**
### Notes:

- Wallmounts are such a pain to setup, and I've completely given up on trying to make it work after hours of wasted time, so as a result; you cannot interact or highlight them even though there are coded interactions for them. If someone else knows how to fix them, they can gladly do it themselves.
- You can check the time by right-clicking the clock then examining it from there. If you attempt to shift click it, it will just point to the wall or floor behind it.
- The ambience audio is originally from [archive.org](https://archive.org/details/ClockTick) under the creative commons license and has been edited by me to clear all the noise and make it shorter.

### Changelog

CL: [New] Adds clocks that can tell time.

### CODE REVIEW

File: UnityProject/AddressablePackingProjects/SoundAndMusic/Assembly-CSharp.csproj [hunk 1]
```diff
@@ -25,8 +25,8 @@
     <DebugSymbols>true</DebugSymbols>
     <DebugType>full</DebugType>
     <Optimize>false</Optimize>
-    <OutputPath>Temp\Bin\Debug\</OutputPath>
-    <DefineConstants></DefineConstants>
+    <OutputPath>Temp\Bin\Debug\Unity.Rider.Editor\</OutputPath>
```
Comment (corp-0) [inline UnityProject/AddressablePackingProjects/SoundAndMusic/Assembly-CSharp.csproj:28]: is this change intentional?

File: UnityProject/Assets/Scripts/Managers/InGameTimeManager.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+		public DateTime UtcTime { get; private set; }
+		public Action OnUpdateTime;
+		private const int GAME_YEAR = 2562;
+
+
+		public override void Awake()
+		{
+			base.Awake();
+			EventManager.AddHandler(Event.RoundStarted, ServerSetupUniversalSpaceTime);
+			UtcTime = DateTime.UtcNow.AddYears(GAME_YEAR);
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Managers/InGameTimeManager.cs:21]: wouldn't this mean ``2022 + 2562`` ?

File: UnityProject/Assets/Scripts/Tilemaps/Behaviours/Objects/RegisterObject.cs [hunk 3]
```diff
@@ -144,7 +146,7 @@ private bool CheckPassableExclusions(GameObject context)
 
 	#region UI Mouse Actions
 
-	public void OnHoverStart()
+	private void OnHoverStart()
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Tilemaps/Behaviours/Objects/RegisterObject.cs]: what is the need for these private methods? Why not just add the body of these into the implementation of the interface?

File: UnityProject/Assets/Scripts/Objects/Wallmounts/ClockWallmount.cs [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+namespace Objects.Wallmounts
+{
+	public class ClockWallmount : NetworkBehaviour, IExaminable, IHoverTooltip, IEmpAble, ICheckedInteractable<HandApply>
+	{
+		[SyncVar] private DateTime UST;
+
+		[SerializeField] private AddressableAudioSource tickSound;
+
+		private bool messedWith = false;
+		private const float TICK_TIME = 1.75f;
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Objects/Wallmounts/ClockWallmount.cs]: is this because the sound you are playing has two ticks?

File: UnityProject/Assets/Scripts/Managers/InGameTimeManager.cs (hunk 2 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Managers/InGameTimeManager.cs:21]: Yes, isnt the game set in the 4ks?

File: UnityProject/AddressablePackingProjects/SoundAndMusic/Assembly-CSharp.csproj (hunk 1 above)
Comment (MaxIsJoe) [reply UnityProject/AddressablePackingProjects/SoundAndMusic/Assembly-CSharp.csproj:28]: Yes

File: UnityProject/Assets/Scripts/Objects/Wallmounts/ClockWallmount.cs [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+			report.AppendLine($"UTC currently is: {InGameTimeManager.Instance.UtcTime}");
+			return report.ToString();
+		}
+
+		public string HoverTip()
+		{
+			return "There's a small analog screen below the arms that displays the date in more detail.";
+		}
+
+		public string CustomTitle()
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Objects/Wallmounts/ClockWallmount.cs]: it is an stylistic choice, but maybe these could use less space in the class written like this
```cs
public string CustomTile() => null
```

File: UnityProject/Assets/Scripts/Objects/Wallmounts/ClockWallmount.cs (hunk 4 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Objects/Wallmounts/ClockWallmount.cs]: Yup

File: UnityProject/Assets/Scripts/Tilemaps/Behaviours/Objects/RegisterObject.cs (hunk 3 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Tilemaps/Behaviours/Objects/RegisterObject.cs]: Good point, they're currently unneeded so they can be moved directly into that function

File: UnityProject/Assets/Scripts/Objects/Wallmounts/ClockWallmount.cs [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+			TextColor text = new TextColor
+			{
+				Text = "Shift+Left Click: Read time.",
+				Color = IntentColors.Help
+			};
+			interactions.Add(text);
+			return interactions;
+		}
+
+		private void MessWithMagnetTime()
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Objects/Wallmounts/ClockWallmount.cs:75]: shouldn't this method also be decorated with ``Server`` ?

File: UnityProject/Assets/Scripts/Managers/InGameTimeManager.cs (hunk 2 above)
Comment (corp-0) [reply UnityProject/Assets/Scripts/Managers/InGameTimeManager.cs:21]: uhhhhm I forgor. Maybe it would be nice to have it read from server configuration in the future, so servers can set the initial time as they wish.

File: UnityProject/Assets/Scripts/Objects/Wallmounts/ClockWallmount.cs (hunk 6 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Objects/Wallmounts/ClockWallmount.cs:75]: Yeah, so clients don't give wrong times on their end.

**Comment by MaxIsJoe** [comment]
Done 👍

**Review by Bod9001 — APPROVED** [review]
